presto-main/src/main/java/com/facebook/presto/sql/planner/CompilerConfig.java

```diff
@@ -37,4 +40,30 @@ public CompilerConfig setExpressionCacheSize(int expressionCacheSize)
         this.expressionCacheSize = expressionCacheSize;
         return this;
     }
+
+    public int getMaxNodeNumInLogicalPlan()
+    {
+        return this.maxNodeNumInLogicalPlan;
+    }
+
+    @Config("planner.max-node-num-in-logical-plan")
+    @ConfigDescription("Maximum number of nodes in logical plan, throw exception when exceed if node_num_in_logical_plan_exceed_thd_exception_enabled is set true")
+    public CompilerConfig setMaxNodeNumInLogicalPlan(int num)
+    {
+        this.maxNodeNumInLogicalPlan = num;
+        return this;
+    }
+
+    public boolean getEnableMaxNodeInLogicalPlanException()
+    {
+        return this.enableMaxNodeInLogicalPlanException;
+    }
+
+    @Config("planner.node-num-in-logical-plan-exceed-thd-exception-enabled")
```

Comment:
I would call this `planner.plan-node-limit-enabled` (and simliarly rename session properties, variables, etc.).

Reply:
Done